config/bootstrap.php

```diff
@@ -64,6 +64,20 @@
 use Cake\Utility\Inflector;
 use Cake\Utility\Security;
 
+/**
+ * Read .env file if APP_NAME is not set.
+ *
+ * You can remove this block if you do not want to use environment
+ * variables for configuration when deploying.
+ */
+if (!env('APP_NAME') && file_exists(CONFIG . '.env')) {
+    $dotenv = new \josegonzalez\Dotenv\Loader([CONFIG . '.env']);
```

Comment:
This should really be a core repo then IMHO as the skeleton relies on it.
Especially when the skeleton suggests to use .env files by creating them `static::createEnvFile($rootDir, $io);`

Also I am no expert on environment files but would it make sense to prefix all env variables with `CAKEPHP_`